src/api/course/index.ts

```diff
@@ -0,0 +1,894 @@
+        return undefined
+      }
+    } else {
+      debug(`** Unable to parse href: ${href}`)
+      return undefined
+    }
+  }
+
+  private buildDeltaMsg(): any {
+    let values: Array<{ path: string; value: any }> = []
+    const navPath = 'navigation.course'
+
+    debug(this.courseInfo)
+
+    values.push({
+      path: `${navPath}.startTime`,
+      value: this.courseInfo.startTime
+    })
+    values.push({
+      path: `${navPath}.targetArrivalTime`,
+      value: this.courseInfo.targetArrivalTime
+    })
+    values.push({
+      path: `${navPath}.activeRoute.href`,
+      value: this.courseInfo.activeRoute.href
+    })
+    values.push({
+      path: `${navPath}.activeRoute.pointIndex`,
+      value: this.courseInfo.activeRoute.pointIndex
+    })
+    values.push({
+      path: `${navPath}.activeRoute.pointTotal`,
+      value: this.courseInfo.activeRoute.pointTotal
+    })
+    values.push({
+      path: `${navPath}.activeRoute.reverse`,
+      value: this.courseInfo.activeRoute.reverse
+    })
+    this.courseInfo.activeRoute.name &&
+      values.push({
+        path: `${navPath}.activeRoute.name`,
+        value: this.courseInfo.activeRoute.name
+      })
+    values.push({
+      path: `${navPath}.activeRoute.waypoints`,
+      value: this.courseInfo.activeRoute.waypoints
+    })
+
+    values.push({
+      path: `${navPath}.nextPoint.href`,
+      value: this.courseInfo.nextPoint.href
+    })
+    values.push({
+      path: `${navPath}.nextPoint.position`,
+      value: this.courseInfo.nextPoint.position
+    })
+    values.push({
+      path: `${navPath}.nextPoint.type`,
+      value: this.courseInfo.nextPoint.type
+    })
+    values.push({
+      path: `${navPath}.nextPoint.arrivalCircle`,
+      value: this.courseInfo.nextPoint.arrivalCircle
+    })
+
+    values.push({
+      path: `${navPath}.previousPoint.position`,
+      value: this.courseInfo.previousPoint.position
+    })
+    values.push({
+      path: `${navPath}.previousPoint.type`,
+      value: this.courseInfo.previousPoint.type
+    })
+
+    values = values.concat(this.buildV1DeltaMsg())
+
+    return {
+      updates: [
+        {
+          values
+        }
+      ]
+    }
+  }
+
+  private buildV1DeltaMsg(): Array<{ path: string; value: any }> {
+    const values: Array<{ path: string; value: any }> = []
+    const navGC = 'navigation.courseGreatCircle'
+    const navRL = 'navigation.courseRhumbline'
+
+    values.push({
+      path: `${navGC}.activeRoute.href`,
+      value: this.courseInfo.activeRoute.href
+    })
+    values.push({
+      path: `${navRL}.activeRoute.href`,
+      value: this.courseInfo.activeRoute.href
+    })
+
+    values.push({
+      path: `${navGC}.activeRoute.startTime`,
+      value: this.courseInfo.startTime
+    })
+    values.push({
+      path: `${navRL}.activeRoute.startTime`,
+      value: this.courseInfo.startTime
+    })
+
+    values.push({
+      path: `${navGC}.nextPoint.value href`,
```

Comment:
This does not look correct. Missing a period? Same below.

Reply:
fixed